src/cockpit/channels/dbus.py

```diff
@@ -205,17 +244,44 @@ def do_open(self, options):
             if err.errno != errno.EBUSY:
                 raise
 
-        self.ready()
-
-    def add_match(self, rule, handler):
+        if self.name is not None:
+            async def get_ready():
+                async with self.watch_processing_lock:
+                    await self.setup_name_owner_tracking()
+                    if self.owner:
+                        self.ready(unique_name=self.owner)
+                    else:
+                        self.close(problem="not-found")
+            task = asyncio.create_task(get_ready())
+            self.tasks.add(task)
+            task.add_done_callback(self.tasks.discard)
+        else:
+            self.ready()
+
+    def add_signal_handler(self, rule, handler):
+        r = rule.copy()
+        r['type'] = 'signal'
+        if 'sender' not in r and self.name:
+            r['sender'] = self.name
+        if r.get('path_namespace') == "/":
+            del r['path_namespace']
```

Comment:
That deserves a comment.  Did you ever get to the bottom of why this is a problem?  Is it a bug in the broker or something?

Reply:
After looking at the dbus-broker code, I think it might be a bug.

https://github.com/bus1/dbus-broker/blob/92142f321fc80c76ab1cc973ce2defa1039cac4a/src/bus/match.c#L387

`match_string_prefix("/foo", "/", '/', false)` returns false, no?  I check that more carefully.

Reply:
https://github.com/bus1/dbus-broker/issues/309